### PR TITLE
[Support RP] - Deprecated commands for Breaking Change Announcement

### PR DESCRIPTION
**Pull request by RudraSharma93Microsoft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Support RP

**Comment by RudraSharma93Microsoft** [comment]
image

File: src/support/HISTORY.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 Release History
 ===============
 
+1.0.4
+-----
+* Deprecated commands list before new upcoming version
```
Comment (jsntcy) [inline src/support/HISTORY.md:6]:
```suggestion
* Add deprecated message for commands before new upcoming version
```

File: src/support/azext_support/custom.py [hunk 2]
```diff
@@ -17,12 +17,17 @@
...
+    logger.warning("Command group 'az support tickets' will be replaced by 'az support in-subscription tickets'" +
+                   " and 'az support no-subscription tickets'")
+    logger.warning("Parameter 'filters' argument will be replaced by 'filter'")
     if filters is None:
         filters = "CreatedDate ge " + str(date.today() - timedelta(days=7))
     return client.list(top=100, filter=filters)
 
 
 def get_support_tickets(cmd, client, ticket_name=None):
+    logger.warning("Command group 'az support tickets' will be replaced by 'az support in-subscription tickets'" +
```
Comment (jsntcy) [inline src/support/azext_support/custom.py:29]: Please rerun your tests lively to ensure those warning messages won't break customers.

File: src/support/azext_support/_params.py [hunk 3]
```diff
@@ -102,7 +102,8 @@ def load_tickets_argument(self, _):
                    help='Indicates if this requires a 24x7 response from Azure. Default is false.')
         c.argument('partner_tenant_id', help='Partner tenant id for Admin On Behalf ' +
                    'of (AOBO) scenario. In addition to logging in to the customer tenant, logging in to the partner ' +
-                   'tenant (PT) using "az login -t PT --allow-no-subscriptions" is required.')
+                   'tenant (PT) using "az login -t PT --allow-no-subscriptions" is required.',
+                   deprecate_info=c.deprecate())
```
Comment (jsntcy) [inline src/support/azext_support/_params.py:106]: Please run commands manually in command line to check if it's what you want.

File: src/support/HISTORY.md (hunk 1 above)
Comment (RudraSharma93Microsoft) [reply src/support/HISTORY.md]: Done

File: src/support/azext_support/custom.py (hunk 2 above)
Comment (RudraSharma93Microsoft) [reply src/support/azext_support/custom.py:29]: Done

File: src/support/azext_support/_params.py (hunk 3 above)
Comment (RudraSharma93Microsoft) [reply src/support/azext_support/_params.py:106]: Verified its what we want

**Review by necusjz — APPROVED** [review]

**Pull request merged**